pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -210,14 +212,15 @@ public String getSegmentMetadata(
               columnLength = storedDataType.size();
             } else if (columnMetadata.hasDictionary()) {
               // For type of variable width (String, Bytes), if it's stored using dictionary encoding, set the
-              // columnLength as the max
-              // length in dictionary.
+              // columnLength as the max length in dictionary.
               columnLength = columnMetadata.getColumnMaxLength();
             } else if (storedDataType == DataType.STRING || storedDataType == DataType.BYTES) {
               // For type of variable width (String, Bytes), if it's stored using raw bytes, set the columnLength as
-              // the length
-              // of the max value.
-              columnLength = ((String) columnMetadata.getMaxValue()).getBytes(StandardCharsets.UTF_8).length;
+              // the length of the max value.
+              ByteArray maxValueByteArray =
+                  storedDataType == DataType.BYTES ? ((ByteArray) columnMetadata.getMaxValue())
+                      : BytesUtils.toByteArray((String) columnMetadata.getMaxValue());
+              columnLength = maxValueByteArray.length();
```

Comment:
could you add test for this?

Reply:
This code change does not look correct to me. AFAIK, we want to fix couple of problems

- Use of * to allow fetching metadata for all columns instead of individual column. Looks like this has been fixed by encoding * ? Please add tests for the same

- A weird scenario where the use of particular column name (internal) fails the API call for some reason. That column type is STRING and stored type is also STRING (UTF-8 bytes essentially). I don't think this is related to the handling of BYTES / BYTE_ARRAY. While column of data types / stored types as BYTES and BYTE_ARRAY should also be handled, the problem we wanted to fix does not seem to be related to it or at least there is no evidence from debugging. I suggest debugging this more to understand the root cause and then come back on this part of the change

Reply:
Discussed offline with @mqliang 

As suggested in the previous comment, let's not fix the BYTES problem in this PR. We can investigate and fix that separately. The url encoding for (with and without *) and that corner case problem for a particular problem should be fixed (which @mqliang  found out is due to min/max value being null and server throwing NPE). 
Let's make sure to add tests for with and without encoded query string -- everything after columns= can be encoded. I think in this PR, controller is re-encoding. We should remove that